### PR TITLE
Revise sparse docs regarding Sparse Compressed tensors

### PR DESCRIPTION
**Pull request by pearu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82108

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -9074,36 +9074,39 @@ def merge_dicts(*dicts):
...
-    compressed_indices (array_like): One-dimensional array of size
-        size[cdim] + 1 where cdim is 0 or 1 depending on the layout.
-        The last element is the number of non-zeros. This tensor
-        encodes the index in values and plain_indices depending on
-        where the given compressed dimension (row or column)
-        starts. Each successive number in the tensor subtracted by the
-        number before it denotes the number of elements in a given
-        compressed dimension.
+    compressed_indices (array_like): (B+1)-dimensional array of size
+        ``(*batchsize, ncompressed_dims + 1)``.  The last element of
```
Comment (bhosmer) [inline torch/_torch_docs.py:9090]: `ncompressed_dims` per above comment (but again, your call)

**Review by bhosmer — APPROVED** [review]
Really nice work! I left a couple of very minor comments inline - couple of typos and one potential terminology tweak that's totally up to you.

File: docs/source/sparse.rst [hunk 2]
```diff
@@ -370,49 +373,141 @@ assumption that the fill value is negative infinity.
 
 .. See https://github.com/Quansight-Labs/rfcs/tree/pearu/rfc-fill-value/RFC-0004-sparse-fill-value for a new API
 
+.. _sparse-compressed-docs:
+
+Sparse Compressed Tensors
++++++++++++++++++++++++++
+
+Sparse Compressed Tensors represents a class of sparse tensors that
+have a common feature of compressing the indices of certain dimension
```
[the remark this answers is not in the document]
Comment (pearu) [reply docs/source/sparse.rst:382]: yes, fixed.

File: docs/source/sparse.rst [hunk 3]
```diff
@@ -370,49 +373,141 @@ assumption that the fill value is negative infinity.
...
+
+.. note::
+
+   We say that an indices tensor ``compressed_indices`` uses CSR
+   encoding if the following invariants are satisfied:
+
+   - ``compressed_indices`` is a contiguous strided 32 or 64 bit integer tensor
+   - ``compressed_indices`` shape is ``(*batchsize, ncompressed_dims +
+     1)`` where ``ncompressed_dims`` is the number of compressed
+     dimensions (e.g. rows or columns)
```
[the remark this answers is not in the document]
Comment (pearu) [reply docs/source/sparse.rst:408]: Thanks for pointing this out! Indeed, `ncompressed_dims` is misleading. This should represent "the size of the compressed dimension". I'll fix the docs accordingly here but there are places in the program code that should be fixed as well. The same applies to `nplain_dims`.

File: docs/source/sparse.rst [hunk 4]
```diff
@@ -370,49 +373,141 @@ assumption that the fill value is negative infinity.
...
+.. note::
+
+   The batches of sparse CSR tensors are dependent: the number of
+   specified elements in all batches must be the same. This somewhat
+   artifical constraint allows efficient storage of the indices of
+   different CSR batches.
+
+.. note::
+
+   The number of sparse and dense dimensions can be aquired using
```
[the remark this answers is not in the document]
Comment (pearu) [reply docs/source/sparse.rst:476]: Fixed.

File: docs/source/sparse.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (pearu) [reply docs/source/sparse.rst]: Fixed.

File: torch/_torch_docs.py (hunk 1 above)
Comment (pearu) [reply torch/_torch_docs.py]: Fixed.

**Review by bhosmer — APPROVED** [review]
Thank you, looks great!

**Pull request closed without merging**